alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/accessor/DescriptorGlobalConfigUtility.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.descriptor.accessor;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.alert.common.action.ApiAction;
+import com.synopsys.integration.alert.common.descriptor.DescriptorKey;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.persistence.accessor.ConfigurationAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationFieldModel;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
+import com.synopsys.integration.alert.common.persistence.util.ConfigurationFieldModelConverter;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+
+public class DescriptorGlobalConfigUtility {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ConfigurationAccessor configurationAccessor;
+    private final ApiAction apiAction;
+    private final ConfigurationFieldModelConverter configurationFieldModelConverter;
+    private DescriptorKey key;
+    private ConfigContextEnum context;
+
+    public DescriptorGlobalConfigUtility(DescriptorKey descriptorKey, ConfigurationAccessor configurationAccessor, ApiAction apiAction,
+        ConfigurationFieldModelConverter configurationFieldModelConverter) {
+        this.key = descriptorKey;
+        this.context = ConfigContextEnum.GLOBAL;
+        this.configurationAccessor = configurationAccessor;
+        this.apiAction = apiAction;
+        this.configurationFieldModelConverter = configurationFieldModelConverter;
+    }
+
+    public DescriptorKey getKey() {
+        return key;
+    }
+
+    public boolean doesConfigurationExist() {
+        try {
+            return !configurationAccessor.getConfigurationByDescriptorKeyAndContext(key, context).isEmpty();
```

Comment:
You access key and context by referencing the fields directly but in other places in this class you use the getKey() method. Can we make this more consistent? Either access the fields or use the get methods.